client/my-sites/posts/post-controls.jsx

```diff
@@ -1,244 +1,223 @@
 import classNames from 'classnames';
 import { noop } from 'lodash';
 
 /**
  * Internal dependencies
  */
-var config = require( 'config' ),
-	utils = require( 'lib/posts/utils' ),
-	Gridicon = require( 'components/gridicon'),
-	recordEvent = require( 'lib/analytics' ).ga.recordEvent;
-
-module.exports = React.createClass( {
-	displayName: 'PostControls',
-
-	mixins: [ PureRenderMixin ],
-
-	propTypes: {
-		post: React.PropTypes.object.isRequired,
-		editURL: React.PropTypes.string.isRequired,
-		onShowMore: React.PropTypes.func.isRequired,
-		onHideMore: React.PropTypes.func.isRequired,
-		onPublish: React.PropTypes.func,
-		onTrash: React.PropTypes.func,
-		onDelete: React.PropTypes.func,
-		onRestore: React.PropTypes.func,
-	},
-
-	view: function() {
-		recordEvent( 'Posts', 'Clicked View Post' );
-	},
-
-	preview: function() {
-		recordEvent( 'Posts', 'Clicked Preview Post' );
-	},
-
-	edit: function() {
-		recordEvent( 'Posts', 'Clicked Edit Post' );
-	},
-
-	copy: function() {
-		recordEvent( 'Posts', 'Clicked Copy Post' );
-	},
-
-	viewStats: function() {
-		recordEvent( 'Posts', 'Clicked View Post Stats' );
-	},
-
-	buildControls: function( controlsArray ) {
-		var postId = this.props.post.ID,
-			controls;
-
-		controls = controlsArray.map( function( item, i ) {
-			const itemKey = 'controls-' + postId + '-' + i;
-			const disabled = item.disabled || false;
-
-			return (
-				<li className={ classNames( { 'post-controls__disabled': disabled } ) } key={ itemKey }>
-					<a href={ item.href } className={ item.className } onClick={ disabled ? noop : item.onClick } target={ item.target ? item.target : null }>
-						<Gridicon icon={ item.icon } size={ 18 } />
-						<span>{ item.text }</span>
-					</a>
-				</li>
-			);
-		}, this );
-
-		return controls;
-	},
-
-	render: function() {
-		var post = this.props.post,
-			availableControls = [],
-			extraControls = [],
-			parsed, previewURL, statsURL, postControlsClass, mainControls, moreControls, i, moreControlsSpliceIndex;
-
-		// NOTE: Currently Jetpack site posts are not returning `post.capabilities`
-		// and those posts will not have access to post management type controls
-
-		// --- Main Controls (not behind ... more link) ---
-		if ( utils.userCan( 'edit_post', post ) ) {
-			availableControls.push( {
-				text: this.translate( 'Edit' ),
-				className: 'post-controls__edit',
-				href: this.props.editURL,
-				target: null,
-				onClick: this.edit,
-				icon: 'pencil'
+import { isEnabled } from 'config';
+import { ga } from 'lib/analytics';
+import { userCan } from 'lib/posts/utils';
+import Gridicon from 'components/gridicon';
+
+const view = () => ga.recordEvent( 'Posts', 'Clicked View Post' );
+const preview = () => ga.recordEvent( 'Posts', 'Clicked Preiew Post' );
+const edit = () => ga.recordEvent( 'Posts', 'Clicked Edit Post' );
+const copy = () => ga.recordEvent( 'Posts', 'Clicked Copy Post' );
+const viewStats = () => ga.recordEvent( 'Posts', 'Clicked View Post Stats' );
```

Comment:
Maybe some better naming here. Eg `recordViewPost`.

Just fyi that we do have a preference for using redux analytics actions instead of using the analytics lib directly. This helps for any future component testing, since analytics assumes window is present.

We can leave connecting the component as a follow up.
See: https://github.com/Automattic/wp-calypso/tree/master/client/state/analytics